### PR TITLE
Fix step 12 JWT signing payload

### LINKED ISSUE #85: Step 12 - Incorrect JWT signing
The shape of the object passed to the JWT `sign` method changes in step 12, this causes the `/user` endpoint to fail.

### PR DESCRIPTION
**Pull request by luke88jones**
fixes #85

### CODE REVIEW

File: src/step-12-database/test/unit/login.test.js [hunk 1]
```diff
@@ -109,4 +109,27 @@ test('POST /login', async t => {
...
+      body: {
+        username: 'alice',
+        password: 'alice',
+      },
+    })
+
+    t.equal(fastify.jwt.sign.callCount, 1)
+    t.same(fastify.jwt.sign.getCall(0).args[0], {
+      username: 'alice'
+    })
```
Comment (simoneb) [inline src/step-12-database/test/unit/login.test.js:133]: minor: let's prefer a more readable way to make these assertions. check out https://sinonjs.org/releases/latest/assertions/

**Review by simoneb — COMMENTED** [review]
See 1 inline comment. Also, does this change need to propagate to other steps too?

File: src/step-12-database/test/unit/login.test.js (hunk 1 above)
Comment (luke88jones) [reply src/step-12-database/test/unit/login.test.js:133]: Ah nice, I've not used tap before so was wondering how to get the sinon-chai `.to.have.been.calledwith()` type assertions.

Comment (simoneb) [reply src/step-12-database/test/unit/login.test.js:133]: you can simply use sinon assertions. if they fail they will just result in an exception and the tap test will fail. tap works with any assertion library basically, you don't have to use the built-in assertions (which btw are quite limited)

**Review by simoneb — APPROVED** [review]

**Pull request merged**